diesel/src/pg/types/uuid.rs

```diff
@@ -0,0 +1,69 @@
+extern crate uuid;
+
+use std::io::prelude::*;
+use std::error::Error;
+
+use expression::AsExpression;
+use expression::bound::Bound;
+use pg::Pg;
+use query_source::Queryable;
+use types::{self, ToSql, IsNull, FromSql};
+
+primitive_impls!(Uuid -> (self::uuid::Uuid, pg: (2950, 2951)));
+
+impl FromSql<types::Uuid, Pg> for self::uuid::Uuid {
+    fn from_sql(bytes: Option<&[u8]>) -> Result<Self, Box<Error>> {
+        match bytes {
+            Some(bytes) => {
+                match self::uuid::Uuid::from_bytes(&bytes) {
+                    Some(u) => Ok(u),
+                    None => {
+                        let message = "Error parsing uuid";
+                        Err(message.into())
+                    }
+                }
+            },
+            None => {
+                let message = "Cannot deserialize an empty uuid";
+                Err(message.into())
+            }
+        }
+    }
+}
+
+impl ToSql<types::Uuid, Pg> for self::uuid::Uuid {
+    fn to_sql<W: Write>(&self, out: &mut W) -> Result<IsNull, Box<Error>> {
+        out.write_all(self.as_bytes())
+            .map(|_| IsNull::No)
+            .map_err(|e| Box::new(e) as Box<Error>)
+    }
+}
+
+#[test]
+fn uuid_to_sql() {
+    let mut bytes = vec![];
+    let test_uuid = uuid::Uuid::from_fields(4_294_967_295, 65_535, 65_535, b"abcdef12");
+    ToSql::<types::Uuid, Pg>::to_sql(&test_uuid, &mut bytes).unwrap();
+    assert_eq!(bytes, test_uuid.as_bytes());
+}
+
+#[test]
+fn some_uuid_from_sql() {
+    let input_uuid =
+        uuid::Uuid::from_fields(4_294_967_295, 65_535, 65_535, b"abcdef12");
+    let output_uuid =
+        FromSql::<types::Uuid, Pg>::from_sql(Some(input_uuid.as_bytes())).unwrap();
+    assert_eq!(input_uuid, output_uuid);
+}
+
+#[test]
+#[should_panic(expected = "Error parsing uuid")]
```

Comment:
Do you think it makes sense to have these tests just make an assertion on the `Err` that comes out, rather than testing the panic with `unwrap`?